tests/app/storage/test_redis.py

```diff
@@ -1,214 +1,208 @@
-            user_id="someuser",
-            session_data="somedata",
-            expires_at=EXPIRES_AT,
-        )
-
-        # When
-        with mock.patch(
-            "app.storage.storage.StorageModel.expiry_field",
-            new_callable=mock.PropertyMock,
-        ) as mock_expiry_field:
-            mock_expiry_field.return_value = None
-            self.redis.put(eq_session)
-
-        # Then
-        expires_in = self.mock_client.ttl(eq_session.eq_session_id)
-        assert expires_in == -1
-
-
-class TestRedisConnectionErrors(AppContextTestCase):
-    def setUp(self):
-        super().setUp()
-        self.mock_client = fakeredis.FakeStrictRedis()
-        self.redis = Redis(self.mock_client)
-
-    def test_put_handles_connection_error_once(self):
-        # Given
-        used_at = datetime.now(tz=timezone.utc)
-        expires_at = used_at + timedelta(seconds=60)
-        jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
-
-        self.redis.client.set = mock.Mock(
-            side_effect=[RedisConnectionError, RedisConnectionError]
-        )
-
-        # When
-        with self.assertRaises(RedisConnectionError):
-            self.redis.put(jti, overwrite=False)
-
-        # Then
-        assert self.redis.client.set.call_count == 2
-
-    def test_get_handles_connection_error_once(self):
-        # Given
-        self.redis.client.get = mock.Mock(
-            side_effect=[RedisConnectionError, RedisConnectionError]
-        )
-
-        # When
-        with self.assertRaises(RedisConnectionError):
-            self.redis.get(EQSession, "test")
-
-        # Then
-        assert self.redis.client.get.call_count == 2
-
-    def test_delete_handles_connection_error_once(self):
-        # Given
-        used_at = datetime.now(tz=timezone.utc)
-        expires_at = used_at + timedelta(seconds=60)
-        jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
-
-        self.redis.client.delete = mock.Mock(
-            side_effect=[RedisConnectionError, RedisConnectionError]
-        )
-
-        # When
-        with self.assertRaises(RedisConnectionError):
-            self.redis.delete(jti)
-
-        # Then
-        assert self.redis.client.delete.call_count == 2
+def test_put_jti_stores_empty_value(redis, redis_client):
+    used_at = datetime.now(tz=timezone.utc)
+    expires_at = used_at + timedelta(seconds=60)
+
+    jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
+
+    redis.put(jti, overwrite=False)
+
+    stored_data = redis_client.get(jti.jti_claim)
+
+    assert b"" == stored_data
+
+
+def test_duplicate_put_jti_fails(redis):
+    used_at = datetime.now(tz=timezone.utc)
+    expires_at = used_at + timedelta(seconds=60)
+
+    jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
+
+    redis.put(jti, overwrite=False)
+
+    with pytest.raises(ItemAlreadyExistsError):
+        redis.put(jti, overwrite=False)
+
+
+def test_put_session(redis):
+    # given
+    eq_session = EQSession(
```

Comment:
Similar to above, EQSession looks the same in each, so extracted to this file of conftest